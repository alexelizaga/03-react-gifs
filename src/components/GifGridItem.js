import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url}) => {
  return (
    <div data-testid='GifGridItem' className='card animate__animated animate__fadeIn'>
      <img data-testid='GifGridItemImg' src={url} alt={title} />
      <p data-testid='GifGridItemTitle'>{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}