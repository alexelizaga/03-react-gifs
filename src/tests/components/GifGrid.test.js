import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test <GifGrid />', () => {

  const category = 'One punch';

  test('should render <GifGrid /> correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const tree = renderer
      .create(<GifGrid category={category} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should show items when its load images', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://www.test.com',
      title: 'Title'
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    // render(<GifGrid category={category} />);
    const tree = renderer
      .create(<GifGrid category={category} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});