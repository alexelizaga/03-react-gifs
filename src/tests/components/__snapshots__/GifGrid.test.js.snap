// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> should render <GifGrid /> correctly 1`] = `
<div
  data-testid="GifGrid"
>
  <h3
    className="animate__animated animate__fadeIn"
    data-testid="GifGridTitle"
  >
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
    data-testid="GifGridLoading"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</div>
`;
